Engine/Resource/Resource.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+
+namespace AntEngine.Resource
+{
+    /// <summary>
+    /// Represents a resource type.
+    /// </summary>
+    public class Resource
+    {
+        public Resource(String type)
+        {
+            Type = type;
+        }
+
+        public String Type { get; private set; }
```

Comment:
I thought we would make Resource the Entity but if we want to have two classes, we should have a Name attribute in the Resource so that the ResourceEntity have the name of its resource type.

Maybe we could also make the Name property of Entity virtual so that he can be overriden in its subclasses ?